Add DIN information. (#1833)
MetaMask/metamask-docs: services/index.md

- **Broad access to major networks** - Infura supports the major networks, allowing you to take advantage of Ethereum's smart
  contracts, IPFS's distributed file system, or high performing layer 2 networks.

> - [**Decentralized Infrastructure Network (DIN) support**](https://www.infura.io/solutions/decentralized-infrastructure-service)

Reviewer: ```suggestion
- [**Decentralized Infrastructure Network (DIN) support**](https://www.infura.io/solutions/decentralized-infrastructure-service) -
```

    DIN works alongside Infura to provide the following:
  - **Failover support for APIs** - Currently available on select networks for customers on Growth or Custom plans; if an Infura API
      endpoint becomes unavailable for any reason, requests can be forwarded to a DIN partner to fulfill the request.
      This ensures that your application remains running, providing uninterrupted service to your users.
  - **Expanded network access** - Infura can extend its network offerings by leveraging DIN to access networks that it doesn't
      natively support.
  - **Expanded method support** - This includes access to debug or trace methods not natively

>       supported by Infura. For these particular types of calls, Infura utilizes DIN to provide user access.

Reviewer: ```suggestion
      supported by Infura. For these types of calls, Infura leverages DIN to provide user access.
```

(I don't really like "leverages" *or* "utilizes," but "leverage" is already used elsewhere in this list so we can stick with that.)

 - **Archive data querying** - [Access historical state data](concepts/archive-data.md) at any given

>       block height. This is an invaluable feature for those who need to perform deep analyses of past transactions, contract

Reviewer: ```suggestion
      block height. This is essential for performing deep analyses of past transactions, contract
```
